src/haddock/libs/libalign.py

```diff
@@ -596,113 +627,272 @@ def align_strct(reference, model, output_path, lovoalign_exec=None):
 
     Parameters
     ----------
-    reference : PosixPath or :py:class:`haddock.libs.libontology.PDBFile`
+    top_aln : Bio.Align.PairwiseAlignments
+        alignment object
+    
+    ref_ch : str
+        reference chain
+    """
+    aln_fname = Path(output_path, f"blosum62_{ref_ch}.aln")
+    log.debug(f"Writing alignment to {aln_fname.name}")
+    with open(aln_fname, "w") as fh:
+        fh.write(str(top_aln))
+    return aln_fname
 
-    model : PosixPath or :py:class:`haddock.libs.libontology.PDBFile`
 
-    output_path : Path
+def sequence_alignment(seq_ref, seq_model):
+    """
+    Perform a sequence alignment.
 
+    Parameters
+    ----------
+    seq_ref : str
+        reference sequence
+    
+    seq_model : str
+        model sequence
+    
     Returns
     -------
-    align_dic : dict
-        dictionary of sequence alignments (one per chain)
+    identity : float
+        sequence identity
+    
+    top_aln : Bio.Align.PairwiseAlignments
+        alignment object
+    
+    aln_ref_seg : tuple
+        aligned reference segment
+    
+    aln_mod_seg : tuple
+        aligned model segment
     """
-    seqdic_ref = pdb2fastadic(reference)
-    seqdic_model = pdb2fastadic(model)
-
-    if seqdic_ref.keys() != seqdic_model.keys():
-        # TODO: Implement chain-matching here
-        return False
-
-    align_dic = {}
-    for ref_chain, model_chain in zip(seqdic_ref, seqdic_model):
-
-        if ref_chain != model_chain:
-            raise AlignError(
-                f"Chain mismatch: {ref_chain} != {model_chain}"
-                )
-
-        align_dic[ref_chain] = {}
-
-        seq_ref = Seq("".join(seqdic_ref[ref_chain].values()))
-        seq_model = Seq("".join(seqdic_model[model_chain].values()))
-
-        aligner = Align.PairwiseAligner()
-        aligner.substitution_matrix = substitution_matrices.load("BLOSUM62")
-        alns = aligner.align(seq_ref, seq_model)
-        top_aln = alns[0]
-
-        aln_fname = Path(output_path, f"blosum62_{ref_chain}.aln")
-        log.debug(f"Writing alignment to {aln_fname.name}")
-        with open(aln_fname, "w") as fh:
-            fh.write(str(top_aln))
-        aligned_ref_segment, aligned_model_segment = top_aln.aligned
-
-        # this should always be true
-        assert len(aligned_ref_segment) == len(aligned_model_segment)
-
-        identity = (
-            str(top_aln).count("|") / float(min(len(seq_ref), len(seq_model)))
-            ) * 100
-
-        if not any(e for e in top_aln.aligned):
+    aligner = Align.PairwiseAligner()
+    aligner.substitution_matrix = substitution_matrices.load("BLOSUM62")
+    alns = aligner.align(seq_ref, seq_model)
+    top_aln = alns[0]
+
+    aln_denom = min(len(seq_ref), len(seq_model))
+    identity = (str(top_aln).count("|") / float(aln_denom)) * 100
+    
+    aln_ref_seg, aln_mod_seg = top_aln.aligned
+    # this should always be true
+    assert len(aln_ref_seg) == len(aln_mod_seg)
```

Comment:
This will raise a  non-specific `AssertionError`, it's best to replace this with a custom exception and do the proper error handling

Reply:
corrected!